### PR TITLE
Plugin Configs

### PR DESCRIPTION
**Pull request by inyono**
This is nearly done, feel free to check out the API locally and report any issues you find
- [x] Fix build
- [x] Fix tests
- [x] Move plugin themes into plugin configs
- [x] Pass sensible plugin registry to rows and text
- [x] Review `bundle-size`
- [x] Review `demo`
- [x] Review `fixtures`
- [x] Review `renderer-ssr`

### CODE REVIEW

**Comment by inyono** [comment]
@Knorrke: Ready for review now 🎉 Gonna check out the build failure later

File: packages/plugins/multimedia-explanation/src/index.ts [hunk 1]
```diff
@@ -1,36 +1,43 @@
...
+import { MultimediaExplanationEditor } from './editor'
 
-export const multimediaExplanationState = (multimediaPlugins: PluginRegistry) =>
+const createMultimediaExplanationState = (
+  config: MultimediaExplanationConfig
+) =>
   object({
-    explanation: child('rows'),
-    multimedia: child(multimediaPlugins[0].name),
+    explanation: child({ plugin: 'rows' }),
```
[the remark this answers is not in the document]
Comment (inyono) [reply packages/plugins/multimedia-explanation/src/index.ts]: I didn't add configs for everything that might make sense to configure. Feel free to adapt the configs in future pull requests :)

**Comment by inyono** [comment]
I have no idea why the build fails... Did even a clean clone of the repository and the build worked quite fine -.-

File: packages/plugins/text/src/plugins/plugin-suggestions.tsx [hunk 2]
```diff
@@ -13,23 +13,18 @@ function mapPlugins(pluginClosure: SlatePluginClosure, editor: Editor) {
...
       .filter(plugin => {
-        if (plugin.name === name || plugin.name === 'rows') return false
         if (!search.length) return true
-
-        return (
-          plugin.title &&
-          plugin.title.toLowerCase().startsWith(search.toLowerCase())
-        )
+        const value = plugin.title || plugin.name
+        console.log(search, value)
```
Comment (Knorrke) [inline packages/plugins/text/src/plugins/plugin-suggestions.tsx]:
```suggestion
```

File: packages/public/core/src/document/renderer.tsx [hunk 3]
```diff
@@ -22,10 +25,18 @@ export function DocumentRenderer({ id, pluginProps }: DocumentProps) {
...
   }
 
+  const defaultConfig =
+    typeof plugin.config === 'function' ? plugin.config(theme) : plugin.config
+  const overrideConfig = (pluginProps && pluginProps.config) || {}
+  const config = R.mergeDeepRight(defaultConfig, overrideConfig)
+
   const pluginState = plugin.state.init(document.state, () => {})
+
+  // This should be fine as soon as we remove DeprecatedPlugin
```
Comment (Knorrke) [inline packages/public/core/src/document/renderer.tsx]: ?

**Pull request merged**